bambi/plots/plot_cap.py

```diff
@@ -0,0 +1,253 @@
+    terms = {}
+    if model._design.common:
+        terms.update(model._design.common.terms)
+
+    if model._design.group:
+        terms.update(model._design.group.terms)
+
+    # Get default values for each variable in the model
+    for term in terms.values():
+        if hasattr(term, "components"):
+            for component in term.components:
+                name = component.name
+                if name not in data_dict:
+                    # For numeric predictors, select the mean.
+                    if component.kind == "numeric":
+                        data_dict[name] = np.mean(data[name])
+                    # For categoric predictors, select the most frequent level.
+                    elif component.kind == "categoric":
+                        data_dict[name] = mode(data[name])
+
+    cap_data = pd.DataFrame(data_dict)
+
+    # Make sure new types are same types than the original columns
+    for column in cap_data:
+        cap_data[column] = cap_data[column].astype(data[column].dtype)
+    return cap_data
+
+
+def plot_cap(model, idata, covariates, hdi_prob=0.94, legend=True, ax=None):
+    """Plot Conditional Adjusted Predictions
+
+    Parameters
+    ----------
+    model : bambi.Model
+        The model for which we want to plot the predictions.
+    idata : arviz.InferenceData
+        The InferenceData object that contains the samples from the posterior distribution of
+        the model.
+    covariates : list
+        A sequence of one or two names of variables. The first variable is taken as the main
+        variable. If present, the second variable is a grouping variable.
+    hdi_prob : float, optional
+        The probability for the credibility intervals. Must be between 0 and 1. Defaults to 0.94.
+    legend : bool, optional
+        Whether to automatically include a legend in the plot. Defaults to ``True``.
+    ax : matplotlib.axes._subplots.AxesSubplot, optional
+        A matplotlib axes object. If None, this function instantiates a new axes object.
+        Defaults to ``None``.
+
+    Returns
+    -------
+    matplotlib.figure.Figure, matplotlib.axes._subplots.AxesSubplot
+        A tuple with the figure and the axes.
+
+    Raises
+    ------
+    ValueError
+        When ``level`` is not within 0 and 1.
+        When the main covariate is not numeric or categoric.
+    """
+
+    covariates = listify(covariates)
+    assert len(covariates) in [1, 2]
+
+    cap_data = create_cap_data(model, covariates)
+    idata = model.predict(idata, data=cap_data, inplace=False)
+
+    if not 0 < hdi_prob < 1:
+        raise ValueError(f"'level' must be greater than 0 and smaller than 1. It is {hdi_prob}.")
+
+    lower_bound = round((1 - hdi_prob) / 2, 4)
+    upper_bound = 1 - lower_bound
+
+    y_hat = idata.posterior[f"{model.response.name}_mean"]
+    y_hat_mean = y_hat.mean(("chain", "draw"))
+    y_hat_bounds = y_hat.quantile(q=(lower_bound, upper_bound), dim=("chain", "draw"))
```

Comment:
Use `az.hdi` and pass hdi_prob directly to it. Additionally, we could have the option to use quantiles or HDI, but still HDI should probably be the default.

Reply:
Agree with this

Reply:
What are the advantages of HDI? I recall this discussion https://github.com/arviz-devs/arviz/discussions/2021 where HDI can result in an unexpected result.

Reply:
Consistency, we use HDI everywhere is ArviZ. Having both is probably a better option.